### PR TITLE
18369 agm extension evaluation

### PR DESCRIPTION
**Pull request by PaulGarewal**
*Issue #:* /bcgov/entity#18369

*Description of changes:*

- added logic to the AGMextension evaluation for eligible and not eligible
-  added unit testing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/views/AgmExtension.vue
Comment (severinbeauvais) [inline src/views/AgmExtension.vue:66]: You don't need to pass this in -- it's already part of the data object on line 65.

File: src/views/AgmExtension.vue
Comment (severinbeauvais) [inline src/views/AgmExtension.vue:67]: You don't use this in the sub-component.

File: src/components/AgmExtension/AgmExtensionEvaluation.vue
Comment (severinbeauvais) [inline src/components/AgmExtension/AgmExtensionEvaluation.vue:112]: I was rather hoping that THIS component could figure these strings instead of having them passed in.

It's the idea of separation of concerns: don't involve extra components in computations if they don't need to know. In this case, the parent component is simply a framework for the data, not the source of these particular props. Karim's component does all the computations. And THIS component can react to Karim's properties and display the proper strings.

Another way to look at it is trying to make self-contained components with as-simple-as-possible interfaces to the rest of their world (props/events).

File: src/interfaces/agm-ext-eval-interface.ts
Comment (severinbeauvais) [inline src/interfaces/agm-ext-eval-interface.ts:3]: When I merge my PR #571 , you'll see I added this to the end of the props list. Doesn't matter either way.

I also initialized it to False instead of null below, meaning no-one has to update it for it to be correct initially. This doesn't really matter since it will probably be set initially by Karim's component logic.

File: src/interfaces/agm-ext-eval-interface.ts
Comment (severinbeauvais) [inline src/interfaces/agm-ext-eval-interface.ts:21]: Both NaN and null as valid "non-numeric" values for a number, so your change is fine.

PS You can check is a variable is a number with `isNaN(x)`.

image

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [inline tests/unit/AgmExtensionEvaluation.spec.ts:116]: Please also verify that the yellow box is displayed when not eligible.

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [inline tests/unit/AgmExtensionEvaluation.spec.ts:17]: Your component doesn't use the store, so you can remove this (and the ones below) and all references to the store above.

File: src/views/AgmExtension.vue
Comment (severinbeauvais) [reply src/views/AgmExtension.vue]: Removed, thanks.

File: src/views/AgmExtension.vue
Comment (severinbeauvais) [reply src/views/AgmExtension.vue]: Removed, thanks.

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [reply tests/unit/AgmExtensionEvaluation.spec.ts]: Removed, thanks.

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [reply tests/unit/AgmExtensionEvaluation.spec.ts:108]: Fixed, thanks.

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [inline tests/unit/AgmExtensionEvaluation.spec.ts:1]: Try to keep the unit tests concise.

Below, you have a lot of duplicate testing.

File: src/components/AgmExtension/AgmExtensionEvaluation.vue
Comment (JazzarKarim) [reply src/components/AgmExtension/AgmExtensionEvaluation.vue:112]: Sev, here's what I was thinking:

This component's non-eligible text will be set in the parent after my logic is done. This is since I believe the text depends on which scenario didn't pass in my component. I might need to follow-up with Yui or Mihai about that though. Not sure if it's the same text for ineligibility for all scenarios or not.

I guess what you're thinking is instead of letting the parent decide the text, we remove these props and have this component set the text depending on whether eligible is true or false? That would work only if the text is the same across all scenarios like I mentioned above though if that makes sense.

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [inline tests/unit/AgmExtensionEvaluation.spec.ts:142]: If the conditional is a `v-if` then check if it exists.

if the conditional is a `v-show` then check if it's visible.

You don't need both here :)

File: src/components/AgmExtension/AgmExtensionEvaluation.vue
Comment (severinbeauvais) [reply src/components/AgmExtension/AgmExtensionEvaluation.vue:112]: Kinda the second one.

For audit purposes, I'm expecting your component to set a value(s) indicating which scenario the business is in (which would get saved in the filing JSON). Then, THIS component can simply look at that value(s) and show the proper text.

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [reply tests/unit/AgmExtensionEvaluation.spec.ts:1]: You could reduce a lot more things that are optional or whatever, but this is OK for now.

File: src/components/AgmExtension/AgmExtensionEvaluation.vue
Comment (JazzarKarim) [reply src/components/AgmExtension/AgmExtensionEvaluation.vue:112]: I see. OK.

@PaulGarewal Can you please follow-up with Yui/Mihai and ask them what would the text be depending on the ineligibility scenario? The UXPin only shows one.

File: src/components/AgmExtension/AgmExtensionEvaluation.vue
Comment (severinbeauvais) [inline src/components/AgmExtension/AgmExtensionEvaluation.vue:70]: I like it. My only request is, could you keep the variables in the same order in the conditionals?

Comment (PaulGarewal) [reply src/components/AgmExtension/AgmExtensionEvaluation.vue:70]: I caught that just now too 😄

File: src/components/AgmExtension/AgmExtensionEvaluation.vue
Comment (severinbeauvais) [reply src/components/AgmExtension/AgmExtensionEvaluation.vue]: Fixed, thanks.

File: src/interfaces/agm-ext-eval-interface.ts
Comment (severinbeauvais) [reply src/interfaces/agm-ext-eval-interface.ts:3]: 571 is merged. Just watch out that the merge doesn't add your prop in a second time.

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [reply tests/unit/AgmExtensionEvaluation.spec.ts]: Fixed, thanks.

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [inline tests/unit/AgmExtensionEvaluation.spec.ts:118]: This is essentially the same test as above, since isEligible=True gives the same results regardless of alreadyExtended.

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [inline tests/unit/AgmExtensionEvaluation.spec.ts:144]: Same as above.

File: tests/unit/AgmExtensionEvaluation.spec.ts
Comment (severinbeauvais) [inline tests/unit/AgmExtensionEvaluation.spec.ts:78]: I think you should pass in extensionDuration and agmDueDate and verify the output of those here.

**Review by severinbeauvais — APPROVED** [review]
Looks good. I have some comments on the unit tests. Please get another approval or 2 before merging. Thanks.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-573-n4nq5i3o.web.app

**Review by jamespaologarcia — APPROVED** [review]
LGTM

**Pull request merged**
